Fix the make build fail on darwin. Fixes openebs/openebs#2204

layog opened the pull request:
Cross platform setting for OS and architecture

Signed-off-by: Ankit Goyal <ankit.goyal@hackerrank.com>

**What this PR does / why we need it**: Cross platform make build

**Which issue this PR fixes**: fixes openebs/openebs#2204

**Special notes for your reviewer**: I tested it on Mac, needs testing for linux once

Pensu commented on buildscripts/agent/build.sh:
```diff
@@ -24,8 +24,27 @@ VERSION="`cat $GOPATH/src/github.com/openebs/maya/VERSION`"
+fi
 
 XC_ARCHS=(${XC_ARCH// / })
 XC_OSS=(${XC_OS// / })
```
Do we still need these values? I think now XC_ARCH and XC_OS will only be one value i.e. the platform the binary is being built upon.

Pensu replied:
Do we still need these values? I think now XC_ARCH and XC_OS will only be one value i.e. the platform the binary is being built upon. Similarly for other changes.

layog commented:
There is some [issue with Github](https://status.github.com/messages) today. I am not able to view your comments. I'll update it later.

layog commented:
There is some [issue with Github](https://status.github.com/messages) today. I am not able to view your comments. I'll update it later.

layog commented:
There is some [issue with Github](https://status.github.com/messages) today. I am not able to view your comments. I'll update it later.

layog commented:
There is some [issue with Github](https://status.github.com/messages) today. I am not able to view your comments. I'll update it later.

layog commented:
There is some [issue with Github](https://status.github.com/messages) today. I am not able to view your comments. I'll update it later.

layog commented:
There is some [issue with Github](https://status.github.com/messages) today. I am not able to view your comments. I'll update it later.

layog commented:
There is some [issue with Github](https://status.github.com/messages) today. I am not able to view your comments. I'll update it later.

layog commented:
There is some [issue with Github](https://status.github.com/messages) today. I am not able to view your comments on the PR. I'll update it later.

layog commented:
There is some [issue with Github](https://status.github.com/messages) today. I am not able to view your comments on the PR. I'll update it later.

layog commented:
@prateekpandey14 Is there a reason that `cstor-volume-mgmt` contains build instructions for `cstor-volume-grpc`?
https://github.com/openebs/maya/blob/7a71a4265c8ee582b021349763c3e28884eb9004/buildscripts/cstor-volume-mgmt/build.sh#L81-L94

prateekpandey14 commented:
@layog yes `cstor-volume-grpc` is part of `cstor-volume-mgmt` build process

layog commented:
@prateekpandey14 Can this be decoupled? I was writing the common `build.sh`, and if we call that for `cstor-volume-grpc`, will that work?

layog commented on buildscripts/agent/build.sh:
```diff
@@ -24,8 +24,27 @@ VERSION="`cat $GOPATH/src/github.com/openebs/maya/VERSION`"
+fi
 
 XC_ARCHS=(${XC_ARCH// / })
 XC_OSS=(${XC_OS// / })
```
I guess this was added for some future requirement. But, if required I can remove this from the common `build` script that I just committed.

layog commented:
@prateekpandey14 I have build the common build script, though I have not removed the earlier ones. I have checked the `Dockerfile`s, none of them seems to make use of `build.sh`, but I was not sure.

layog commented:
@prateekpandey14 I've removed the old build scripts and the build is passing.

@Pensu I have also removed the array based building as there can never be multiple OS or arch being built at the same time.

Pensu commented on buildscripts/build.sh:
```diff
@@ -0,0 +1,139 @@
+    -o $output_name\
+    ./cmd/${CTLNAME}
+if [ "${PNAME}" = "cstor-volume-mgmt" ]; then
+    env GOOS=$GOOS GOARCH=$GOARCH go build -ldflags \
```
Do we need GOOS and GOARCH, if we are only building the binary for the platform where we are running the buildscript from. Please correct me if my understanding is wrong.

layog replied:
Yeah, we can directly use `XC_OS` and `XC_ARCH`, without declaring `GOOS` and `GOARCH`

layog commented on buildscripts/build.sh:
```diff
@@ -0,0 +1,139 @@
+GOARCH="${XC_ARCH}"
+output_name="bin/${PNAME}/"$GOOS"_"$GOARCH"/"$CTLNAME
+
+if [ $GOOS = "windows" ]; then
```
Yeah, as of now, this won't be executed, since we are asserting that the supported architectures are `linux` and `darwin`.
